nextcloudappstore/core/views.py

```diff
@@ -18,19 +20,48 @@ def get_context_data(self, **kwargs):
 class CategoryAppListView(ListView):
     model = App
     template_name = 'app/list.html'
+    allow_empty = True
 
     def get_queryset(self):
         category_id = self.kwargs['id']
         queryset = super().get_queryset()
+
         if category_id:
-            return queryset.filter(categories__id=category_id)
-        else:
-            return queryset
+            queryset = queryset.filter(categories__id=category_id)
+
+        if self.search_terms:
+            query = None
+
+            for term in self.search_terms:
+                q = Q(translations__name__contains=term) | \
+                    Q(translations__description__contains=term)
+                if query is None:
+                    query = q
+                else:
+                    query = query | q
+
+            queryset = queryset.filter(query)
+
+            # Remove duplicates that for some reason sometimes occur
+            queryset = list(set(queryset))
+
+        return queryset
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['categories'] = Category.objects.all()
         category_id = self.kwargs['id']
         if category_id:
             context['current_category'] = Category.objects.get(id=category_id)
+        if self.search_terms:
+            context['search_query'] = ' '.join(self.search_terms)
+            context['search_terms'] = self.search_terms
```

Comment:
The `search_terms` context data is not used anywhere. Do we keep it in case we need a list of search terms in a template, or do we remove it?